src/test/java/com/google/sps/EventTest.java

```diff
@@ -0,0 +1,129 @@
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.Event;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class EventTest {
+
+  // time class constants
+  private final static int HOUR = 60;
+  private final static int HALF_HOUR = 30;
+  private final static int NINETY_MIN = 90;
+  private final static int HOUR_FIFTY_FIVE = 115;
+  private final static int THREE_HOURS = 180;
+
+  // files for event 1
+  private static final String GOLDENGATEPARK = "GGPark";
+  private static final String ADDRESS =  "4265 24th Street San Francisco, CA, 94114";
+  private static final String DATE = "2020-06-25";
+  private static final int START_TIME = 1000;
+  private static final int DEFAULT_TRAVEL_TIME = 30;
+  private static final int DURATION_IN_MIN = 80;
+  private static final String STR_START_TIME = "2020-06-25T10:00:00";
+  private static final String STR_MANUAL_END_TIME = "2020-06-25T11:20:00";
+  private static final String STR_DEF_END_TIME = "2020-06-25T11:00:00";
+
+  @Test
+  public void testMinuteToHHMMFormatOneHour() {
+    String output = Event.convertToFormat(HOUR);
+    Assert.assertEquals("0100", output);
+  }
+
+  @Test
+  public void testMinuteToHHMMFormatHalfHour() {
+    String output = Event.convertToFormat(HALF_HOUR);
+    Assert.assertEquals("0030", output);
+  }
+
+  @Test
+  public void testMinuteToHHMMFormatNinetyMin() {
+    String output = Event.convertToFormat(NINETY_MIN);
+    Assert.assertEquals("0130", output);
+  }
+
+  @Test
+  public void testMinuteToHHMMFormatHourFivetyFive() {
+    String output = Event.convertToFormat(HOUR_FIFTY_FIVE);
+    Assert.assertEquals("0155", output);
+  }
+
+  @Test
+  public void testMinuteToHHMMFormatThreeHour() {
+    String output = Event.convertToFormat(THREE_HOURS);
+    Assert.assertEquals("0300", output);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testMinuteToHHMMFormatLessThanZero() {
+    Event.convertToFormat(-1);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testMinuteToHHMMFormatGreaterThanFullDay() {
+    Event.convertToFormat(1440);
+  }
+  
+  @Test
+  public void testConstructorManualTimeSpent() {
+    Event e1 = new Event(GOLDENGATEPARK, ADDRESS, DATE, START_TIME, 
+                        DEFAULT_TRAVEL_TIME, DURATION_IN_MIN);
+
+    Assert.assertEquals(GOLDENGATEPARK, e1.getName());
+    Assert.assertEquals(ADDRESS, e1.getAddress());
+    Assert.assertEquals(DATE, e1.getDate());
+    Assert.assertEquals(START_TIME, e1.getStartTime());
+    Assert.assertEquals(DEFAULT_TRAVEL_TIME, e1.getTravelTime());
+    Assert.assertEquals(1120, e1.getEndTime());
+
+    Assert.assertEquals(STR_START_TIME, e1.getStrStartTime());
+    Assert.assertEquals(STR_MANUAL_END_TIME, e1.getStrEndTime());
```

Comment:
nit: this may actually be easier to read if DATE, START_TIME, STR_START_TIME, and STR_MANUAL_END_TIME were not constants.

Styling guidance is pretty different for tests. Eg: in non-test code you'd want to eliminate the redundancy of declaring the same string in multiple places as you have done here. But in a test redundancy is ok if it improves clarity. So in this case if the strings were declared inline I'd have all the information I need to understand this test in one place, eg: I could see that the date part of STR_START_TIME is the same as DATE, etc without scrolling back and forth between this method and the constants declared above. (compare with the ADDRESS constant which is more appropriate since I can see it's given in the constructor and returned unmodified by the getter, so I don't really need to know it's contents in order to see what's going on)

Feel free to disregard this comment for now, I'll see if I can find some documentation on test styling that provides a bit more context

Reply:
Ok! I thought so too but I thought styling said that I needed to declare all constants. LMK if you find anything and I'll change it!

Reply:
Yea, styling for test code can be pretty different. Here's some documentation that another host put together: http://go/step-tott especially check out the 'Tests Too DRY? Make them DAMP' document which focuses on the redundancy vs legibility trade-off for tests.